### PR TITLE
Paginate lessons

### PR DESCRIPTION
**Pull request by andylolz**
I’ve been using this bit of code for ages, but haven’t
PRed it. I guess because it’s super hacky. Anyway,
here’s a PR.

### CODE REVIEW

**Comment by martinpeck** [comment]
Thanks @andylolz. I suspect we're going to put this PR on ice for the moment. 

We're in the process of redesigning the projects site at the moment, so I'd rather work the "lessons should be paginated" requirement into that redesign (or not).

**Comment by andylolz** [comment]
That’s fine. I’ll close this one, then.

**Comment by andylolz** [comment]
Tbh this doesn’t force pagination – it just adds the option. I don’t see why it would be a problem to merge in the interim.

**Comment by andylolz** [comment]
Tbh this was/is mergable, and doesn’t force you to use paginated lessons. It just adds the javascript so you _can_, which would be a good way of testing if this is a feature that code club volunteers want.

**Comment by martinpeck** [comment]
I've looked at the PR locally, and it looks like something worth discussing within the team, but my reluctance to merge it was related the following:

- Initial comment suggested that this was "super hacky", so didn't want to spend time on it if that was the case.
- I wasn't sure how "tested" this was in terms of cross browser testing and mobile/desktop.
- We are in the process of reviewing and redesigning the UX for displaying projects. I would rather look at whether paging makes sense there, than here. Particularly because...
- It's hard to remove features that people are using. If we decide that paging isn't important then I'd rather we didn't add feature debt
- The current implementation has limitations...
  - the presence of paging isn't obvious (controls are at bottom of screen, but nothing suggests paging anywhere else)
  - navigating between steps is hard (there's no way to get to step N easily)
  - lots of "back button" required to get out of a lesson (you can't easily jump back to the list of lessons if you're on the last step)
  - autoplaying embedded scratch continues to autoplay when hidden, resulting in music/sounds etc from a hidden step
- Not sure we've had time to think how this would be rolled out if it were found to be useful
  - would it be the default, or an option?
  - if the default, we'd still need to test the pages without pagination for PDF generation

Basically, it didn't feel ready. Given this, I didn't want to merge the changes and then have people find it/use it, take dependencies upon it, and require us to support it.

This feels like something we should have in a test/experimental branch, but not on the main project site.

**Comment by martinpeck** [comment]
I'd be happy to merge it "as is" so long as it's not used (i.e. not used in official links, not advertised, not built upon)

**Comment by andylolz** [comment]
Okay, cool – I don’t mind either way. The argument for not merging sounds totally reasonable.

I only replied because there’s some small amount of effort required on my part to maintain my own fork that includes this, and every time I do it I wish I didn’t have to! So obviously I’d prefer this as an easter egg upstream.

> We're in the process of redesigning the projects site at the moment

Is this happening in the open somewhere?
